src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -648,6 +651,31 @@ internal virtual async Task<SecurityKey> ValidateSignatureAsync(JsonWebToken sig
             throw LogHelper.LogExceptionMessage(new SignedHttpRequestInvalidSignatureException(LogHelper.FormatInvariant(LogMessages.IDX23034, signedHttpRequest.EncodedToken)));
         }
 
+        /// <summary>
+        /// Validates the nonce claim of the signed http request.
+        /// </summary>
+        /// <param name="signedHttpRequest">A SignedHttpRequest.</param>
+        /// <param name="popKey">A Pop key used to validate the signed http request nonce signature.</param>
+        /// <param name="signedHttpRequestValidationContext">A structure that wraps parameters needed for SignedHttpRequest validation.</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns>A PoP <see cref="SecurityKey"/> that validates signature of the <paramref name="signedHttpRequest"/>.</returns>
+        internal virtual bool ValidateNonceAsync(JsonWebToken signedHttpRequest, SecurityKey popKey, SignedHttpRequestValidationContext signedHttpRequestValidationContext, CancellationToken cancellationToken)
```

Comment:
Should it be async? And need clarify the return value and the exception.

Reply:
We probably don't want to ship an actual implementation of this. Just define what the delegate looks like, this signature looks good though.

Reply:
Whoops, misunderstood, this is okay so long as it just calls the delegate on the validationparams

Reply:
Ok, this can be return void and just throw if the delegate returns false.